kubernetes/skaffold/src/main/kotlin/com/google/kubernetes/tools/skaffold/SkaffoldExecutorService.kt

```diff
@@ -116,11 +112,10 @@ abstract class SkaffoldExecutorService {
 
             // the following settings only apply to `init` mode.
             if (settings.executionMode == ExecutionMode.INIT) {
+                // the flags are mutually exclusive, and shouldn't be used together.
                 if (settings.analyzeOnInit) {
```

Comment:
I guess this is ok since we are constructing the commands. But what if instead of arbitrarily favoring `analyzeOnInit` we just fail the execution if both are set?

Reply:
We can do that, and enforce certain command combinations to ensure proper calling - might be useful for future users of the service. We can start with this test case, let me convert this into an exception.